src/back8sup.sh

```diff
@@ -131,3 +132,6 @@ do
   log "INFO done exporting all $KIND in namespace $NS"
 done
 log "INFO done exporting namespace $NS"
+
+log "INFO remove backups older than $EXPIRE days"
+find $DST_FOLDER -mtime +$EXPIRE -type f -delete &>/dev/null
```

Comment:
Did you consider a "keep the last x generations" logic rather than a "remove everything older than x" logic?

Reply:
no. Why is the other logic better?

Reply:
In the "remove everything older than x" case old backups would be removed if the job ever fails for longer than $EXPIRE days. Thus I usually prefer to keep old backup generations and not backups older than x. In the past I've done things like the following:

```bash 
GENERATIONS=30

ls -t -w 1 "$DST_FOLDER/*" | tail -n +$GENERATIONS | xargs rm -f
```

Maybe I'm overthinking this and it's fine as is.

Reply:
This is good, but IMHO we should rather check that we do not run the cleanup command  if the job fails, right?

Reply:
> This is good, but IMHO we should rather check that we do not run the cleanup command if the job fails, right?

sounds fair